### PR TITLE
EVG-2414 Implement ec2 cost functions

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
I think this may make more sense now that I've moved it into the cloud package.

File: cloud/ec2_util.go [hunk 1]
```diff
@@ -610,6 +610,169 @@ func newMakeBlockDeviceMappings(mounts []MountPoint) ([]*ec2aws.BlockDeviceMappi
...
+		os = osWindows
+	}
+	ebsCost, err := m.blockDeviceCosts(instance.BlockDeviceMappings, end.Sub(start))
+	if err != nil {
+		return 0, errors.Wrap(err, "calculating block device costs")
+	}
+	spotCost, err := m.calculateSpotCost(instance, os, start, end)
+	if err != nil {
+		return 0, err
+	}
```
Comment (tychoish) [inline cloud/ec2_util.go]: wrap/withstack errors.

File: cloud/ec2_util.go [hunk 2]
```diff
@@ -610,6 +610,169 @@ func newMakeBlockDeviceMappings(mounts []MountPoint) ([]*ec2aws.BlockDeviceMappi
...
+	start time.Time
+	end   time.Time
+}
+
+// describeHourlySpotPriceHistory talks to Amazon to get spot price history, then
+// simplifies that history into hourly billing rates starting from the supplied
+// start time. Returns a slice of hour-separated spot prices or any errors that occur.
+func (m *ec2Manager) describeHourlySpotPriceHistory(input hourlySpotPriceHistoryInput) ([]spotRate, error) {
+	// expand times to contain the full runtime of the host
+	startFilter, endFilter := input.start.Add(-5*time.Hour), input.end.Add(time.Hour)
```
Comment (tychoish) [inline cloud/ec2_util.go]: why -5 here?

File: cloud/ec2_util.go [hunk 3]
```diff
@@ -610,6 +610,169 @@ func newMakeBlockDeviceMappings(mounts []MountPoint) ([]*ec2aws.BlockDeviceMappi
 	return mappings, nil
 }
 
+func (m *ec2Manager) costForDurationOnDemand(h *host.Host, start, end time.Time) (float64, error) {
```
Comment (tychoish) [inline cloud/ec2_util.go]: I'd move these methods into a different file.

Comment (bsamek) [reply cloud/ec2_util.go]: done

File: cloud/ec2_util.go (hunk 1 above)
Comment (bsamek) [reply cloud/ec2_util.go]: done

File: cloud/ec2_util.go (hunk 2 above)
Comment (bsamek) [reply cloud/ec2_util.go]: I'm not sure. This is from the current implementation. I *think* it was to ensure that we got the whole history, but `-time.Hour` should do it, so changed.

**Pull request merged**